Sources/SpeziDevicesUI/Devices/BatteryIcon.swift

```diff
@@ -0,0 +1,102 @@
+//
+// This source file is part of the Stanford Spezi open-project
+//
+// SPDX-FileCopyrightText: 2024 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import SwiftUI
+
+
+/// Battery Icon with optional label.
+public struct BatteryIcon: View {
+    private let percentage: Int
+    private let isCharging: Bool
+
+
+    public var body: some View {
+        Label {
+            Text(verbatim: "\(percentage) %")
+        } icon: {
+            batteryIcon // hides accessibility, only text will be shown
+                .foregroundStyle(.primary)
+        }
+            .accessibilityRepresentation {
+                if !isCharging {
+                    Text(verbatim: "\(percentage) %")
+                } else {
+                    Text(verbatim: "\(percentage) %, is charging")
+                }
+            }
+    }
+
+
+    @ViewBuilder private var batteryIcon: some View {
+        Group {
+            if isCharging {
+                Image(systemName: "battery.100percent.bolt")
```

Comment:
Interesting that there are no charging icons for lower battery states 🤷‍♂️

Reply:
Also a missed opportunity that this is not a value based icon where the icon automatically adjusts based on the value you pass in (like it is with e.g. wifi symbols).
Interestingly, this view was reused from NAMS. So if that app is updated one can remove it from there.

Reply:
Missed opportunity by Apple.

Nice, that would be great!